### PR TITLE
Include correct list of discriminator information

### PR DESCRIPTION
**Pull request by millicentachieng**
Fixes https://github.com/microsoft/OpenAPI.NET.OData/issues/219

Retrieve list of derived types for structured types and adds it to the mapping property of the discriminator object.

### CODE REVIEW

File: src/Microsoft.OpenApi.OData.Reader/Generator/OpenApiSchemaGenerator.cs [hunk 1]
```diff
@@ -436,23 +436,26 @@ internal static OpenApiSchema CreateSchemaTypeSchema(this ODataContext context,
...
-                    }.Reference.ReferenceV3;
+                            Reference = new OpenApiReference
+                            {
+                                Type = ReferenceType.Schema,
+                                Id = item.FullTypeName()
+                            }
+                        }.Reference.ReferenceV3;
+
+                        mapping.Add("#" + item.FullTypeName(), v3RefIdentifier);
+                    }
```
Comment (baywet) [inline src/Microsoft.OpenApi.OData.Reader/Generator/OpenApiSchemaGenerator.cs:453]:
```CSharp
Dictionary<string, string> mapping = derivedTypes.ToDictionary(x => $"#{x.FullTypeName()}", x => new OpenApiSchema
                        {
                            Reference = new OpenApiReference
                            {
                                Type = ReferenceType.Schema,
                                Id = x.FullTypeName()
                            }
                        }.Reference.ReferenceV3);
```

**Review by baywet — CHANGES_REQUESTED** [review]
thank for taking this on.
On top of the suggestion, please add an entry in the changelog in the csproj.

**Review by baywet — DISMISSED** [review]

**Review by irvinesunday — DISMISSED** [review]

**Review by baywet — APPROVED** [review]

**Review by irvinesunday — APPROVED** [review]

**Comment by baywet** [comment]
@irvinesunday @millicentachieng FYI I just approved the CD, the nuget should be available in 10 minutes

**Comment by irvinesunday** [comment]
> @irvinesunday @millicentachieng FYI I just approved the CD, the nuget should be available in 10 minutes

@baywet, we weren't intending to get `1.0.11-preview1` out yet with just this fix. There were a couple of other fixes that were meant to go out 1st before we do the release.

**Comment by baywet** [comment]
ah... sorry

**Pull request merged**